src/custom-element.js

```diff
@@ -1664,6 +1677,20 @@ function createBaseCustomElementClass(win) {
       return policyId;
     }
 
+    /**
+     * Get the purpose consents that should be granted.
+     * @return {?Array<string>}
+     */
+    getPurposesConsent_() {
+      const purposes = this.getAttribute('data-block-on-consent-purposes');
+      // null or empty string
+      if (!purposes) {
+        return null;
+      }
+      const purposesArray = purposes.split(',');
+      return purposesArray.length ? purposesArray : null;
```

Comment:
Since `purposes` is already guaranteed to be non-empty at this point, `purposesArray.length > 0`

```suggestion
      const purposes = this.getAttribute('data-block-on-consent-purposes');
      // null or empty string
      return purposes ? purposes.split(',') : null;
```